### PR TITLE
FT2-1449 cosmosdb query refactor

### PR DESCRIPTION
**Pull request by andycyberian**

### CODE REVIEW

File: DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs [hunk 1]
```diff
@@ -38,30 +37,20 @@ public class CosmosDbDescribeContentItemHelper : IDescribeContentItemHelper
 
         public async Task<IEnumerable<IQuery<object?>>> GetRelationshipCommands(IDescribeRelationshipsContext context)
         {
-            var currentList = new List<ContentItemRelationship>();
+            var currentList = new List<Tuple<string, string>>();
```
Comment (ryanbarlowmethods) [inline DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs:40]: I wouldnt use this syntax now - nicer to use the (string, string) with optional names

File: DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs [hunk 2]
```diff
@@ -38,30 +37,20 @@ public class CosmosDbDescribeContentItemHelper : IDescribeContentItemHelper
...
 
-            var uniqueCommands = allRelationships
-                .Select(relationship => relationship.RelationshipPathString)
-                .GroupBy(relationshipPathString => relationshipPathString)
-                .Select(relationshipPathStringGroup => relationshipPathStringGroup.First());
+            var uniqueCommands = allRelationships.Distinct();
 
             return uniqueCommands
-                .Select(c => new CosmosDbNodeAndNestedOutgoingRelationshipsQuery(c!)).Cast<IQuery<object?>>().ToList();
+                .Select(c => new CosmosDbNodeAndNestedOutgoingRelationshipsQuery("SELECT * FROM c WHERE c.id = @id0", c.Item1, c.Item2)).Cast<IQuery<object?>>().ToList();
```
Comment (ryanbarlowmethods) [inline DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs:53]: if you name them you can make this nice

File: DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbSubgraphQuery.cs [hunk 3]
```diff
@@ -86,28 +91,9 @@ public Query Query
...
-#pragma warning restore CS8602 // Dereference of a possibly null reference.
-#pragma warning restore IDE0008 // Use explicit type
-
-#pragma warning disable CS8620 // Argument cannot be used for parameter due to differences in the nullability of reference types.
-                return new Query(
-                    @$"select * from c where c.id = '{id}'",
-                    parameters);
-#pragma warning restore CS8620 // Argument cannot be used for parameter due to differences in the nullability of reference types.
+                var contentType = NodeLabels.First(nodeLabel => !nodeLabel.Equals("Resource", StringComparison.InvariantCultureIgnoreCase)).ToLower();
+                var id = IdPropertyValue.ToString()?.Split('/').Last();
```
Comment (ryanbarlowmethods) [inline DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbSubgraphQuery.cs:95]: I think we can use the getcontenttypeandid method?

File: DFC.ServiceTaxonomy.GraphSync/GraphSyncers/VisualiseGraphSyncer.cs [hunk 4]
```diff
@@ -419,8 +419,7 @@ private static string GetContHasName(Dictionary<string, object> properties, stri
         {
             var detailCommand = new List<IQuery<object?>>
             {
-                new CosmosDbNodeAndNestedOutgoingRelationshipsQuery(
-                    $"select * from c where c.id in ({string.Join(',', ids)})|{contentType}")
+                new CosmosDbNodeAndNestedOutgoingRelationshipsQuery("SELECT * FROM c WHERE c.id in (@id0)", string.Join(',', ids), contentType)
```
Comment (ryanbarlowmethods) [inline DFC.ServiceTaxonomy.GraphSync/GraphSyncers/VisualiseGraphSyncer.cs:422]: Is this right? looks a bit odd - i imaging @id0 gets expanded? is it just the name thats weird?

File: DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbSubgraphQuery.cs (hunk 3 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbSubgraphQuery.cs]: amended to use getcontenttypeandid()

File: DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs (hunk 1 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs:40]: Replaced with shorthand tuple

File: DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs (hunk 2 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbDescribeContentItemHelper.cs:53]: Replaced with shorthand tuple

File: DFC.ServiceTaxonomy.GraphSync/GraphSyncers/VisualiseGraphSyncer.cs (hunk 4 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.GraphSync/GraphSyncers/VisualiseGraphSyncer.cs]: Have added an additional parameter to support custom parameter names

**Review by ryanbarlowmethods — APPROVED** [review]

**Pull request merged**